programs/uxd/tests/program_uxd/integration_tests/test_credix.rs

```diff
@@ -0,0 +1,84 @@
+use solana_program_test::processor;
+use solana_program_test::tokio;
+use solana_program_test::ProgramTest;
+use solana_program_test::ProgramTestContext;
+use solana_sdk::signer::keypair::Keypair;
+use solana_sdk::signer::Signer;
+
+use crate::program_credix;
+use crate::program_spl;
+use crate::program_uxd;
+
+const PAYER: usize = 0;
+const UXD_AUTHORITY: usize = 1;
+const CREDIX_AUTHORITY: usize = 3;
+
+#[tokio::test]
+async fn test_credix() -> Result<(), String> {
+    let keypairs = [
+        Keypair::new(),
+        Keypair::new(),
+        Keypair::new(),
+        Keypair::new(),
+        Keypair::new(),
+        Keypair::new(),
+        Keypair::new(),
+    ];
+
+    let mut program_test = ProgramTest::default();
+
+    program_test.add_program("uxd", uxd::id(), processor!(uxd::entry));
+
+    /*
+    program_test.add_account_with_file_data(
+        Pubkey::from_str("GMiQHsRQpdbgaRA3Y2SJUxC7wBvBoCFpKFHCnMHM4f8a").unwrap(),
+        1_000_000_000_000,
+        keypairs[CREDIX_AUTHORITY].pubkey(),
+        "tests/fixtures/credix_client.so",
+    );
+     */
+    program_test.prefer_bpf(true);
+    program_test.add_program("credix_client", credix_client::id(), None);
+
+    let mut program_test_context: ProgramTestContext = program_test.start_with_context().await;
+
+    for keypair in &keypairs {
+        program_spl::instructions::process_lamports_airdrop(
+            &mut program_test_context,
+            &keypair.pubkey(),
+            1_000_000_000_000,
+        )
+        .await?;
+    }
+
+    let collateral_mint = Keypair::new();
+    program_spl::instructions::process_token_mint_init(
```

Comment:
Namespace and the multiple indirection might make it confusing, as it feels it's not our domain but it is. But anyway you got the ownership on ship I'm not the main maintainer of the codebase anymore.

Overall good adaptation of tests for the codebase! gg

Reply:
Yeah I understand, but given the massive difficulty of making things work, having a library of function we know just simply work is very helpful instead of having to lookup random repos or undocumented solana functions, just having "known to work instructions" is super helpful

That was the only way to keep my sanity when trying to find what works and what doesnt

Reply:
That being said,  I may definitly have been a bit overboard with it, feel free to let me know if I could reduce the amount of indirection in some places where its excessive